### ISSUE TITLE
Memory issue in v1.10.6

### ISSUE THREAD

**Issue by pirvudoru** [rmosolgo/graphql-ruby#2888]
**Describe the bug**

After updating from 1.10.5 to 1.10.6 we noticed that the memory being allocated on our production servers kept growing making the servers unresponsive. (memory leak)

**Versions**

`graphql` version: 1.10.6
`rails` (or other framework): 6.0.2.2

**GraphQL schema**
```
  class Schema < GraphQL::Schema
    query Types::Query
    use GraphQL::Tracing::AppsignalTracing

    rescue_from(...) { 'Not found' }
    rescue_from(...) { 'Not found' }
    rescue_from(...) { |e| "Bad request: #{e.message}" }

    def self.resolve_type(_type, obj, _ctx)
      Schema.types[obj.class.name.split('::').last]
    end
  end
```

Query Type looks like:

```
module Types
  class Query < GraphQL::Schema::Object
    graphql_name 'Query'

    field :foo, resolver: Resolvers::FooResolver
    # ...
  end
end
```

**Steps to reproduce**
- setup rails
- install graphql gem v1.10.5
- setup graphql controller
- controller test that executed same graphql query 1000x
- use get_process_mem to print out memory `GetProcessMem.new.mb` before running and after running
- install graphql gem v1.10.6
- repeat test

**Expected behavior**
- memory usage should remain constant after update

**Actual behavior**
- memory increases by 140mb for 1000 runs (my usecase) after the update

**Comment by pirvudoru** [comment]
I may be already late to the party. Just saw https://github.com/rmosolgo/graphql-ruby/issues/2878

i'll test that update.

**Comment by rmosolgo** [comment]
Duplciate of #2878 , fixed in #2884 and released in 1.10.7 yesterday. Please try the latest version and let me know if you run into any more trouble!
